packages/synapse-react-client/test/lib/containers/personal_access_token/CreateAccessTokenModal.test.tsx

```diff
@@ -25,6 +25,23 @@ function renderComponent(props: CreateAccessTokenModalProps) {
   })
 }
 
+function setUp(props: CreateAccessTokenModalProps) {
+  const user = userEvent.setup()
+  const component = renderComponent(props)
+  const tokenInput = screen.getByRole('textbox')
+  const checkboxes = {
+    view: screen.getByRole('checkbox', { name: 'View' }),
+    modify: screen.getByRole('checkbox', { name: 'Modify' }),
+    download: screen.getByRole('checkbox', { name: 'Download' }),
+  }
+  const initialButtons = {
+    close: screen.getByRole('button', { name: /close/i }),
+    cancel: screen.getByRole('button', { name: /cancel/i }),
+    create: screen.getByRole('button', { name: /create token/i }),
+  }
+  return { component, user, tokenInput, checkboxes, initialButtons }
```

Comment:
For less dynamic components, I really like this style of getting all of the elements that you'll want to interact with in `setUp`. Will definitely be using this myself when possible